src/main/java/com/hibernate/cinema/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,41 @@
+package com.hibernate.cinema.security;
+
+import com.hibernate.cinema.model.User;
+import com.hibernate.cinema.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+    private final PasswordEncoder encoder;
+
+    @Autowired
+    public CustomUserDetailsService(UserService userService, PasswordEncoder encoder) {
+        this.userService = userService;
+        this.encoder = encoder;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email);
```

Comment:
let's return Optional in the method findByEmail, and get the value here with orElseThrow

Reply:
In the previous weeks, it was said that we need to return optional only from dao layer. Can I just use UserDao here then?